test/parallel/xyk-pallet.API.pool.soloToken.test.ts

```diff
@@ -0,0 +1,280 @@
+    Assets.mintToken(token1, testUser, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(token2, testUser, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(token3, testUser, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser),
+    Sudo.sudoAs(
+      testUser,
+      Xyk.createPool(
+        token2,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token3,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    )
+  );
+
+  liqId = await getLiquidityAssetId(token2, token3);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(token1.toNumber(), 20),
+    Assets.promotePool(liqId.toNumber(), 20)
+  );
+});
+
+beforeEach(async () => {
+  [testUser1] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(liqId, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser1)
+  );
+
+  testUser1.addAsset(token1);
+  testUser1.addAsset(MGA_ASSET_ID);
+});
+
+test("GIVEN pool and solo token AND both were created at the same time AND the activated amounts are similar THEN the available rewards are the same AND when the user claims, it gets the same amounts", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(liqId, BN_BILLION)),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqId)),
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(token1))
+  );
+
+  const rewardsSolo = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  const rewardsPool = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqId
+  );
+
+  expect(rewardsSolo.rewardsAlreadyClaimed).bnEqual(
+    rewardsPool.rewardsAlreadyClaimed
+  );
+});
+
+test("GIVEN pool and solo token AND both were created at the same time AND the activated amounts are similar AND then some new activation happens after one session and some deactivation THEN the poolRewards storage is the same", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(liqId, BN_BILLION)),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(liqId, BN_MILLION)),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_MILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.deactivateLiquidity(liqId, BN_MILLION)),
+    Sudo.sudoAs(testUser1, Xyk.deactivateLiquidity(token1, BN_MILLION))
+  );
+
+  const rewardsSolo = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  const rewardsPool = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqId
+  );
+
```

Comment:
What happens with the other storage items? why only validating pool ratio and not yet claimed?

Reply:
I think that we need to check the properties that were changed. But I changed it

Reply:
Plz, add assertions for all the storage entries.